tests/rename-deps.rs

```diff
@@ -0,0 +1,174 @@
+        "#)
+        .file("src/lib.rs", "")
+        .build();
+
+    assert_that(p.cargo("build").masquerade_as_nightly_cargo(),
+                execs().with_status(101)
+                       .with_stderr("\
+error: failed to parse manifest at `[..]`
+
+Caused by:
+  feature `dependencies-as` is required
+
+consider adding `cargo-features = [\"dependencies-as\"]` to the manifest
+"));
+}
+
+#[test]
+fn empty_array_bad() {
+    Package::new("bar", "0.1.0").publish();
+    Package::new("bar", "0.2.0").publish();
+
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            cargo-features = ["dependencies-as"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [dependencies]
+            bar = []
+        "#)
+        .file("src/lib.rs", "")
+        .build();
+
+    assert_that(p.cargo("build").masquerade_as_nightly_cargo(),
+                execs().with_status(101)
+                       .with_stderr("\
+error: failed to parse manifest at `[..]`
+
+Caused by:
+  cannot specify an empty list of dependency specifications for `bar`
+"));
+}
+
+#[test]
+fn one_element_ok() {
+    Package::new("bar", "0.1.0").publish();
+
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            cargo-features = ["dependencies-as"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [dependencies]
+            bar = [
+                { version = "0.1.0" }
+            ]
+        "#)
+        .file("src/lib.rs", "
+            extern crate bar;
+        ")
+        .build();
+
+    assert_that(p.cargo("build").masquerade_as_nightly_cargo(),
+                execs().with_status(0));
+}
+
+#[test]
+fn array_of_strings_bad() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            cargo-features = ["dependencies-as"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [dependencies]
+            bar = ["0.1"]
+        "#)
+        .file("src/lib.rs", "")
+        .build();
+
+    assert_that(p.cargo("build").masquerade_as_nightly_cargo(),
+                execs().with_status(101)
+                       .with_stderr("\
+error: failed to parse manifest at `[..]`
+
+Caused by:
+  invalid type: string \"0.1\", expected struct DetailedTomlDependency for \
+  key `dependencies.bar`
+"));
+}
+
+#[test]
+fn rename_dependency() {
+    Package::new("bar", "0.1.0").publish();
+    Package::new("bar", "0.2.0").publish();
+
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            cargo-features = ["dependencies-as"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [dependencies]
+            bar = [
+                { version = "0.1.0" },
+                { version = "0.2.0", as = "baz" },
+            ]
```

Comment:
Hm the syntax does not look too natural to me. Is the following a valid TOML?

```TOML
[dependencies]
bar = "0.1.0"
bar = { version = "0.2.0", as = "baz"}
```

Reply:
Unfortunately TOML-the-spec doesn't allow for something like that b/c it's counted as a redefinition of an existing key :(